src/db.c

```diff
@@ -2195,9 +2195,17 @@ int expireIfNeeded(redisDb *db, robj *key, int flags) {
 int dbExpand(const redisDb *db, uint64_t db_size, dbKeyType keyType, int try_expand) {
     dict *d;
     if (server.cluster_enabled) {
+        /* We don't know exact number of keys that would fall into each slot, but we can
+         * approximate it, assuming even distribution, divide it by the number of slots. */
+        int slots = 0;
+        for (int i = 0; i < CLUSTER_SLOTS; i++) {
+            if (clusterNodeGetSlotBit(server.cluster->myself, i))
+                slots++;
+        }
+        if (slots > 0) db_size = db_size / slots;
```

Comment:
I remember this change was added by @vitarb and somewhere during refactoring it got lost. Thanks for adding it.

I'm also not sure when this could happen. Should we add this then,  `if (slots == 0) slots = CLUSTER_SLOTS`

Reply:
If `slots == 0`, this function will not do anything (`clusterNodeGetSlotBit` will again return 0 every time in the for loop below) so we can just return immediately.

```C
        if (slots == 0) return C_OK;
        db_size = db_size / slots;
```